cylc/rose/rose.py

```diff
@@ -48,34 +57,108 @@ def get_rose_vars(dir_=None, opts=None):
             }
 
     TODO:
-        - Once the CLI for the ``rose suite-run`` replacement command is
-          ready plumb in the the equivelent of
-          ``rose suite-run --opt-conf-key=""``.
         - Consider allowing ``[jinja2:flow.conf]`` as an alias for
           consistency with cylc.
     """
     config = {
         'env': None,
-        'empy:suite.rc': None,
-        'jinja2:suite.rc': None
+        'template variables': None,
+        'templating detected': None
     }
-    # Return None if dir_ does not exist
+    # Return a blank config dict if dir_ does not exist
     if not rose_config_exists(dir_):
         return config
 
-    # Load the config tree
+    # Load the raw config tree
     config_tree = rose_config_tree_loader(dir_, opts)
 
-    # For each of the template language sections...
-    for section in ['jinja2:suite.rc', 'empy:suite.rc', 'env']:
-        if section in config_tree.node.value:
-            config[section] = dict(
-                [
-                    (item[0][1], item[1].value) for
-                    item in config_tree.node.value[section].walk()
-                ]
-            )
-
+    templating = None
+    if (
+        'jinja2:suite.rc' in config_tree.node.value and
+        'empy:suite.rc' in config_tree.node.value
+    ):
+        raise MultipleTemplatingEnginesError(
+            "You should not define both jinja2 and empy in the same "
+            "configuration file."
+        )
+    elif 'jinja2:suite.rc' in config_tree.node.value:
+        templating = 'jinja2:suite.rc'
+    elif 'empy:suite.rc' in config_tree.node.value:
+        templating = 'empy:suite.rc'
```

Comment:
```suggestion
        templating = 'empy'
```

May be easier to merge them here into a single section called `template variables`. Better still updated `template variables` if it exists so we can move people from `*:suite.rc` to `template variables` for the future.